apps/api/src/app/events/e2e/digest-events.e2e.ts

```diff
@@ -927,8 +1046,133 @@ describe('Trigger event - Digest triggered events - /v1/events/trigger (POST)',
       _environmentId: session.environment._id,
       _templateId: template._id,
     });
+    const allJobsBackoff = await jobRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+    });
+    expect(jobCount).to.equal(15);
+  });
+
+  it('should create multiple digests based on different nested digestKeys with backoff', async function () {
+    const postId = MessageRepository.createObjectId();
+    const postId2 = MessageRepository.createObjectId();
+    template = await session.createTemplate({
+      steps: [
+        {
+          type: StepTypeEnum.DIGEST,
+          content: '',
+          metadata: {
+            unit: DigestUnitEnum.MINUTES,
+            amount: 5,
+            digestKey: 'nested.postId',
+            type: DigestTypeEnum.BACKOFF,
+            backoffUnit: DigestUnitEnum.MINUTES,
+            backoffAmount: 5,
+          },
+        },
+        {
+          type: StepTypeEnum.CHAT,
+          content: 'Hello world {{nested.postId}}' as string,
+        },
+      ],
+    });
+    await triggerEvent({
+      customVar: 'first',
+      nested: { postId: postId },
+    });
+    await session.awaitParsingEvents();
+
+    await triggerEvent({
+      customVar: 'fourth',
+      nested: { postId: postId },
+    });
+    await session.awaitParsingEvents();
+
+    await triggerEvent({
+      customVar: 'fifth',
+    });
+
+    await session.awaitParsingEvents();
+
+    await triggerEvent({
+      customVar: 'second',
+      nested: { postId: postId2 },
+    });
+
+    await session.awaitParsingEvents();
+
+    await triggerEvent({
+      customVar: 'third',
+      nested: { postId: postId2 },
+    });
+    await session.awaitParsingEvents();
+
+    await triggerEvent({
+      customVar: 'sixth',
+    });
+    await session.awaitRunningJobs(template?._id, false, 3);
+
+    let digests = await jobRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+      type: StepTypeEnum.DIGEST,
+    });
+
+    expect(digests[0].payload.nested.postId).not.to.equal(digests[1].payload.nested.postId);
+    expect(digests[2].payload?.nested?.postId).to.equal(undefined);
+
+    expect(digests.length).to.equal(3);
+
+    digests = await jobRepository.find({
+      _environmentId: session.environment._id,
+      _templateId: template._id,
+      type: StepTypeEnum.DIGEST,
+    });
+
+    for (const digest of digests) {
+      await runJob.execute(
+        RunJobCommand.create({
+          jobId: digest._id,
+          environmentId: digest._environmentId,
+          organizationId: digest._organizationId,
+          userId: digest._userId,
+        })
+      );
+    }
```

Comment:
I am still unsure that you would need to execute this to accomplish the goal of the test. What are the reasons behind forcing the job executions for digest type of jobs?